File: sydney/Kernel/Common/Request.cpp
// -*-Mode: C++; tab-width: 4; c-basic-offset: 4;-*-
// vi:set ts=4 sw=4:
//
// Request.cpp -- リクエストをあわらすクラス
// 
// Copyright (c) 2002, 2007, 2009, 2010, 2023 Ricoh Company, Ltd.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// 

namespace {
const char srcFile[] = __FILE__;
const char moduleName[] = "Common";
}

#include "SyDefault.h"
#include "SyReinterpretCast.h"
#include "Common/Request.h"
#include "Common/ClassID.h"

#include "ModUnicodeString.h"

_TRMEISTER_USING

using namespace Common;

//
//	FUNCTION public
//	Common::Request::Request -- コンストラクタ(1)
//
//	NOTES
//	コンストラクタ。
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
Request::Request()
: m_iCommand(Undefined)
{
}

//
//	FUNCTION public
//	Common::Request::Request -- コンストラクタ(2)
//
//	NOTES
//	コンストラクタ。
//
//	ARGUMENTS
//	int iCommand_
//		リクエスト番号
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
Request::Request(int iCommand_)
: m_iCommand(iCommand_)
{
}

//
//	FUNCTION public
//	Common::Request::~Request -- デストラクタ
//
//	NOTES
//	デストラクタ。
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
Request::~Request()
{
}

//
//	FUNCTION public
//	Common::Request::setCommand -- リクエスト番号を設定する
//
//	NOTES
//	リクエスト番号を設定する。
//
//	ARGUMENTS
//	int iCommand_
//		リクエスト番号
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	なし
//
void
Request::setCommand(int iCommand_)
{
	m_iCommand = iCommand_;
}

//
//	FUNCTION public
//	Common::Request::getCommand -- リクエスト番号を得る
//
//	NOTES
//	リクエスト番号を得る。
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	int
//		リクエスト番号
//
//	EXCEPTIONS
//	なし
//
int
Request::getCommand() const
{
	return m_iCommand;
}

//
//	FUNCTION public
//	Common::Request::serialize -- シリアル化を行う
//
//	NOTES
//	シリアル化を行う。
//
//	ARGUMENTS
//	ModArchive& cArchiver_
//		アーカイバ
//
//	RETURN
//	なし
//
//	EXCEPTIONS
//	その他
//		下位の例外はそのまま再送
//
void
Request::serialize(ModArchive& cArchiver_)
{
	if (cArchiver_.isStore())
	{
		//書出し
		cArchiver_ << m_iCommand;
	}
	else
	{
		//読出し
		cArchiver_ >> m_iCommand;
	}
}

//
//	FUNCTION public
//	Common::Request::getClassID -- クラスIDを得る
//
//	NOTES
//	クラスIDを得る。
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	int
//		クラスID
//
//	EXCEPTIONS
//	なし
//
int
Request::getClassID() const
{
	return ClassID::RequestClass;
}

//
//	FUNCTION public
//	Common::Request::toString -- 文字列で取り出す(for Debug)
//
//	NOTES
//	デバッグ用に文字列で取り出す。
//
//	ARGUMENTS
//	なし
//
//	RETURN
//	ModString
//		デバッグ文字列
//
//	EXCEPTIONS
//	その他
//		下位の例外はそのまま再送
//
ModUnicodeString
Request::toString() const
{
	ModUnicodeString s("Request:");

	switch(m_iCommand)
	{
	case BeginConnection:	s += "BeginConnection";		break;	//コネクション開始
	case EndConnection:		s += "EndConnection";		break;	//コネクション終了
	case BeginSession:		s += "BeginSession";		break;	//セッション開始
	case EndSession:		s += "EndSession";			break;	//セッション終了
	case BeginWorker:		s += "BeginWorker";			break;	//ワーカ開始
	case CancelWorker:		s += "CancelWorker";		break;	//ワーカ中断

	case Shutdown:			s += "Shutdown";			break;	//シャットダウン

	case ExecuteStatement:	s += "ExecuteStatement";	break;	//SQL文の実行
	case PrepareStatement:	s += "PrepareStatement";	break;	//SQL文のコンパイル
	case ExecutePrepare:	s += "ExecutePrepare";		break;	//コンパイル結果の実行
	case ErasePrepareStatement:	s += "ErasePrepareStatement";	break;	//コンパイル結果の削除

	case ReuseConnection:	s += "ReuseConnection";		break;	//コネクションを再利用する
	case NoReuseConnection:	s += "NoReuseConnection";	break;	//コネクションを再利用しない

	case CheckAvailability:	s += "CheckAvailability";	break;	// 利用可能性チェック

	case PrepareStatement2:	s += "PrepareStatement2";	break;	//SQL文のコンパイル
	case ErasePrepareStatement2: s += "ErasePrepareStatement2"; break;//最適化結果を削除する

	case BeginSession2:		s += "BeginSession2";		break;	// Starting session with user management
	case EndSession2:		s += "EndSession2";			break;	// Ending session with user management

	case CreateUser:		s += "CreateUser";			break;	// Creating new user
	case DropUser:			s += "DropUser";			break;	// Deleting a user
	case ChangeOwnPassword:	s += "ChangeOwnPassword";	break;	// Changing own password
	case ChangePassword:	s += "ChangePassword";		break;	// Changing others password

	case Shutdown2:			s += "Shutdown2";			break;	//サーバの終了

	case Sync:				s += "Sync";				break;	//同期

	case QueryProductVersion: s += "QueryProductVersion"; break; // Investigating product version(JDBC)

	default:				s += "Undefine";			break;	//未定義
	}
	return s;
}

//
//	Copyright (c) 2002, 2007, 2009, 2010, 2023 Ricoh Company, Ltd.
//	All rights reserved.
//

File: sydney/Java/JDBC/src/jp/co/ricoh/doquedb/common/Request.java
// -*-Mode: Java; tab-width: 4; c-basic-offset: 4;-*-
// vi:set ts=4 sw=4:
//
// Request.java -- リクエストをあらわすクラス
//
// Copyright (c) 2002, 2003, 2006, 2007, 2009, 2023 Ricoh Company, Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

package jp.co.ricoh.doquedb.common;

/**
 * リクエストをあらわすクラス
 *
 */
public final class Request
	implements Serializable
{
	/** コネクション開始 */
	public final static int BEGIN_CONNECTION = 1;
	/** コネクション終了 */
	public final static int END_CONNECTION = 2;
	/** セッション開始 */
	public final static int BEGIN_SESSION = 3;
	/** セッション終了 */
	public final static int END_SESSION = 4;
	/** ワーカ開始 */
	public final static int BEGIN_WORKER = 5;
	/** ワーカ中断 */
	public final static int CANCEL_WORKER = 6;

	/** サーバ終了 */
	public final static int SHUTDOWN = 7;

	/** SQL文の実行 */
	public final static int EXECUTE_STATEMENT = 8;
	/** SQL文のコンパイル */
	public final static int PREPARE_STATEMENT = 9;
	/** コンパイル結果の実行 */
	public final static int EXECUTE_PREPARE_STATEMENT = 10;
	/** コンパイル結果の削除 */
	public final static int ERASE_PREPARE_STATEMENT = 11;

	/** コネクションを再利用する */
	public final static int REUSE_CONNECTION = 12;
	/** コネクションを再利用しない */
	public final static int NO_REUSE_CONNECTION = 13;

	/** 利用可能性チェック */
	public final static int CHECK_AVAILABILITY = 14;

	/** SQL文のコンパイル */
	public final static int PREPARE_STATEMENT2 = 15;
	/** コンパイル結果の削除 */
	public final static int ERASE_PREPARE_STATEMENT2 = 16;

	/** セッション開始(ユーザーつき) */
	public final static int BEGIN_SESSION2 = 17;
	/** セッション終了(ユーザーつき;未使用) */
	public final static int END_SESSION2 = 18;

	/** ユーザーの追加 */
	public final static int CREATE_USER = 19;
	/** ユーザーの削除 */
	public final static int DROP_USER = 20;
	/** パスワードの変更(自分自身) */
	public final static int CHANGE_OWN_PASSWORD = 21;
	/** パスワードの変更(他人) */
	public final static int CHANGE_PASSWORD = 22;
	/** サーバの終了 */
	public final static int SHUTDOWN2 = 23;

	/** 同期を取る */
	public final static int SYNC = 101;

	/** ProductVersionを問い合わせる */
	public final static int QUERY_PRODUCT_VERSION = 201;

	/** 不明なリクエスト */
	public final static int UNDEFINED = -1;

	/** 利用可能性のチェック対象：サーバー */
	public final static int AVAILABILITY_TARGET_SERVER = 0;

	/** 利用可能性のチェック対象：データベース */
	public final static int AVAILABILITY_TARGET_DATABASE = 1;

	/** 値 */
	private int _request;

	/**
	 * 新たにリクエストを作成する。
	 */
	public Request()
	{
		_request = UNDEFINED;
	}

	/**
	 * 新たにリクエストを作成する。
	 *
	 * @param request_	格納する値
	 */
	public Request(int request_)
	{
		_request = request_;
	}

	/**
	 * リクエストを得る
	 *
	 * @return	格納されているリクエスト
	 */
	public int getRequest()
	{
		return _request;
	}

	/**
	 * リクエストを設定する
	 *
	 * @param request_	格納するリクエスト
	 */
	public void setRequest(int request_)
	{
		_request = request_;
	}

	/**
	 * ストリームから読み込む
	 *
	 * @param input_	入力用のストリーム
	 * @throws java.io.IOException
	 *			入出力関係の例外が発生した
	 * @see Serializable#readObject(InputStream) readObject
	 */
	public void readObject(InputStream input_)
		throws java.io.IOException
	{
		_request = input_.readInt();
	}

	/**
	 * ストリームに書き出す
	 *
	 * @param output_	出力用のストリーム
	 * @throws java.io.IOException
	 *			入出力関係の例外が発生した
	 * @see	Serializable#writeObject(OutputStream) writeObject
	 */
	public void writeObject(OutputStream output_)
		throws java.io.IOException
	{
		output_.writeInt(_request);
	}

	/**
	 * {@link ClassID クラスID}を得る
	 *
	 * @return {@link ClassID クラスID}
	 * @see Serializable#getClassID() getClassID
	 */
	public int getClassID()
	{
		return ClassID.REQUEST;
	}
}

//
// Copyright (c) 2002, 2003, 2006, 2007, 2009, 2023 Ricoh Company, Ltd.
// All rights reserved.
//
